### PR TITLE
Add features to log precision and recall metric for each class. (detection task)

### PR DESCRIPTION
**Pull request by hglee98**
## Description

This PR aims to present features to log detection precision and recall metric for each and total class.

example:
```
+----------------+--------------+----------+----------+------------+-------------+----------+
| Class number   | Class name   | mAP50    | mAP75    | mAP50_95   | Precision   | Recall   |
+================+==============+==========+==========+============+=============+==========+
| 0              | car          | 0.915827 | 0.770286 | 0.634498   | 0.915033    | 0.854962 |
+----------------+--------------+----------+----------+------------+-------------+----------+
| 1              | bus          | 0.74875  | 0.74875  | 0.673875   | 0.75        | 0.6      |
+----------------+--------------+----------+----------+------------+-------------+----------+
| 2              | truck        | 0.889912 | 0.765798 | 0.618583   | 0.736842    | 1        |
+----------------+--------------+----------+----------+------------+-------------+----------+
| 3              | person       | 0.910091 | 0.550151 | 0.528495   | 0.895652    | 0.855956 |
+----------------+--------------+----------+----------+------------+-------------+----------+
| 4              | bicycle      | 0        | 0        | 0          | 0           | 0        |
+----------------+--------------+----------+----------+------------+-------------+----------+
| 5              | motorcycle   | 0.75     | 0.75     | 0.6        | 1           | 0.5      |
+----------------+--------------+----------+----------+------------+-------------+----------+
| -              | Mean         | 0.70243  | 0.597498 | 0.509242   | 0.716254    | 0.635153 |
+----------------+--------------+----------+----------+------------+-------------+----------+
```

Closes: #578

We recommend to link at least one existing issue for PR. Before your create a PR, please check if there is an issue for this change.  

## Change(s)

- Implement detection precision and recall metric
- Add precision and recall metric as a default metric for the object detection task.

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`. 

## Changelog

If your PR is granted to `dev` branch, codeowner will add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file including a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

Please enable **Allow edits and access to secrets by maintainers** so that our maintainers can update the `CHANGELOG.md`.

### CODE REVIEW

File: src/netspresso_trainer/metrics/detection/metric.py [hunk 1]
```diff
@@ -188,6 +188,72 @@ def average_precisions_per_class(
...
+        if total_true == 0:
+            continue
+        if total_predictions == 0:
+            precisions[int(class_id), 0] = 0.
+            continue
+
+        false_positives = (1 - matches[is_class]).sum(0)
+        true_positives = matches[is_class].sum(0)
+        precision = true_positives / (true_positives + false_positives + eps)
+        precisions[int(class_id), 0] = precision[0]
```
Comment (illian01) [inline src/netspresso_trainer/metrics/detection/metric.py:221]: It seems only use precision of 0.5 threshold .
How about return precisions of all threshold and pick only one in `Precision` class?

And I think renaming `Precision` class to `Precision50` may be less confusing.

File: src/netspresso_trainer/metrics/detection/metric.py [hunk 2]
```diff
@@ -188,6 +188,72 @@ def average_precisions_per_class(
...
+
+        if total_true == 0:
+            continue
+        if total_predictions == 0:
+            recalls[int(class_id), 0] = 0.
+            continue
+
+        true_positives = matches[is_class].sum(0)
+        recall = true_positives / (total_true + eps)
+        recalls[int(class_id), 0] = recall[0]
```
Comment (illian01) [inline src/netspresso_trainer/metrics/detection/metric.py:253]: It seems only use recall of 0.5 threshold .
How about return precisions of all threshold and pick only one in `Recall` class?

And I think renaming `Recall` class to `Recall50` may be less confusing.

**Review by illian01 — CHANGES_REQUESTED** [review]
Thank you for adding new metrics of detection task.
I have some code review, please check detailed comments.

File: config/data/local/debug.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+      image: image/val  # directory for valid images
+      label: label/val  # directory for valid labels
+    test:
+      image: image/val #images/val
+      label: label/val #labels/val  # directory for valid labels
+    pattern:
+      image: ~
+      label: ~
+  id_mapping: ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13']  # class names
+  pallete: ~
```
[the remark this answers is not in the document]
Comment (hglee98) [reply config/data/local/debug.yaml]: Thank you for checking this!!

File: src/netspresso_trainer/metrics/detection/metric.py (hunk 2 above)
Comment (hglee98) [reply src/netspresso_trainer/metrics/detection/metric.py]: I push additional commit to fix the issue you've mentioned! Thank you for the review!!

File: src/netspresso_trainer/metrics/detection/metric.py (hunk 1 above)
Comment (hglee98) [reply src/netspresso_trainer/metrics/detection/metric.py]: I push additional commit to fix the issue you've mentioned! Thank you for the review!!

**Review by illian01 — APPROVED** [review]
LGTM!

**Pull request merged**